core/src/main/scala/kafka/common/ClientIdAndBroker.scala

```diff
@@ -26,6 +26,10 @@ trait ClientIdBroker {
 }
 
 case class ClientIdAndBroker(clientId: String, brokerHost: String, brokerPort: Int) extends ClientIdBroker {
+
+  // ipv6 addresses contain ':' which is not a valid character in metrics ObjectName
+  def brokerHostMetricsTag = brokerHost.replace(':', '.')
```

Comment:
I think we do this type of "sanitizing" for multiple metrics and I think its a standard pattern that we would do for any colon (':'). Maybe we should move this into the code for defining a metric. Somewhere under the `newMeter` implementation? It looks like `KafkaMetricsGroup.metricName` already tries to return a "Sanitized metric name object." Maybe that would be the place?